OTAnalytics/plugin_ui/customtkinter_gui/toplevel_flows.py

```diff
@@ -9,53 +10,77 @@
     tk_events,
 )
 from OTAnalytics.plugin_ui.customtkinter_gui.messagebox import InfoBox
+from OTAnalytics.plugin_ui.customtkinter_gui.treeview_template import IdResource
 
-DISTANCE = "Distance"
+FLOW_ID = "Id"
+FLOW_NAME = "Name"
 START_SECTION = "Start section"
 END_SECTION = "End section"
+DISTANCE = "Distance"
 
 
 class ToplevelFlows(CTkToplevel):
     def __init__(
         self,
         title: str,
         initial_position: tuple[int, int],
-        section_ids: list[str],
+        section_ids: list[IdResource],
         input_values: dict | None = {},
         **kwargs: Any,
     ) -> None:
         super().__init__(**kwargs)
         self.title(title)
         self._section_ids = section_ids
+        self._section_name_to_id = self._create_section_name_to_id(section_ids)
+        self._section_id_to_name = self._create_section_id_to_name(section_ids)
+        self._current_name = StringVar()
         self.input_values: dict = self.__create_input_values(input_values)
         self.protocol("WM_DELETE_WINDOW", self.close)
         self._initial_position = initial_position
+        self.__set_initial_values()
         self._get_widgets()
         self._place_widgets()
         self._set_initial_position()
         self._set_focus()
         self._set_close_on_return_key()
 
+    def _create_section_name_to_id(self, sections: list[IdResource]) -> dict[str, str]:
+        return {resource.name: resource.id for resource in sections}
+
+    def _create_section_id_to_name(self, sections: list[IdResource]) -> dict[str, str]:
+        return {resource.id: resource.name for resource in sections}
+
+    def __set_initial_values(self) -> None:
+        self._current_name.set(self.input_values.get(FLOW_NAME, ""))
+
     def __create_input_values(self, input_values: Optional[dict]) -> dict:
         if input_values:
             return input_values
         return {
+            FLOW_ID: "",
+            FLOW_NAME: "",
             START_SECTION: "",
             END_SECTION: "",
             DISTANCE: 0,
         }
 
     def _get_widgets(self) -> None:
+        self.label_id = CTkLabel(master=self, text="Name")
```

Comment:
This should be named `name` instead of `id` now. But I already did this in #199 (OP#2125)

Reply:
Then I will leave it as is to reduce merge conflicts